src/plugins/kibana_react/public/field_icon/field_icon.tsx

```diff
@@ -61,6 +62,7 @@ export const typeToEuiIconMap: Partial<Record<string, IconMapEntry>> = {
   number: { icon: 'number', color: colors[0] },
   _source: { icon: 'editorCodeBlock', color: colors[3] },
   string: { icon: 'string', color: colors[4] },
+  nested: { icon: 'nested', color: colors[4] },
```

Comment:
@cchaos what color would you recommend here?

Reply:
We're working on coming up with permanent field type icon/color combinations on the EUI side, but for now use `colors[2]`